src/app/core/components/left-nav-tree/left-nav-tree.component.html

```diff
@@ -0,0 +1,181 @@
+
+<div class="tree-header" *ngIf="data && data.parent" (click)="navigateToParent()">
+  <span class="tree-header-handle">
+    <i class="fa fa-chevron-left"></i>
+    <div class="icon-folder"><img src="/assets/images/tree-folder.svg" alt=""></div>
+  </span>
+  <span class="tree-header-title">{{ data.parent.title }}</span>
+</div>
+
+<p-tree [value]="nodes" i18n-emptyMessage emptyMessage="There is no content to display">
+
+  <!-- "chapter" type -->
+  <ng-template let-node pTemplate="chapter">
+    <span
+      class="node-tree-item folder"
+      [ngClass]="{ expanded: node.expanded, locked: node.locked, current: node.data && node.data.current }"
+    >
+      <span class="node-item-branch">
+        <span class="top-branch">
+          <span class="mask-branch"></span>
+        </span>
+        <span class="bottom-branch folder-branch">
+          <span class="mask-branch"></span>
+        </span>
+      </span>
+      <span class="node-item-content" [ngClass]="{ unlocked: !node.locked }" (click)="selectNode(node)">
+        <span class="node-select">
+          <alg-score-ring
+            *ngIf="node.data.element.currentScore >= 0; else folderNoScore"
+            [currentScore]="node.data.element.currentScore"
+            [bestScore]="node.data.element.bestScore"
+            [isValidated]="node.data.element.validated"
+            [diameter]="30"
+            [forTree]="true"
+            [isDark]="true"
+            [icon]="node.expanded ? 'folder-open' : 'folder'"
+          ></alg-score-ring>
+          <ng-template #folderNoScore>
+            <i class="fa" [ngClass]="{ 'fa-folder': !node.expanded, 'fa-folder-open': node.expanded }" style="font-size: 1.2rem;"></i>
+          </ng-template>
+        </span>
+        <span class="node-label">
+          <span class="label-container">
+            <span class="node-label-title" [ngClass]="{ locked: node.locked }">
+              {{ node.label }}
+              <span *ngIf="node.data.element.groupName">(via group "{{ node.data.element.groupName }}")</span>
+            </span>
+            <span class="node-state" *ngIf="node.data.status === 'loading'">
+              <p-progressSpinner [style]="{ width: '3rem', height: '3rem', 'margin-top': '-0.5rem' }"></p-progressSpinner>
+            </span>
+            <span class="node-state" *ngIf="node.data.status === 'error'">
+              <i class="fa fa-exclamation-triangle" style="font-size: 1.2rem;"></i>
```

Comment:
maybe move style in scss file because there is quite a few repetition in this file (for the p-progressSpinner and icons)

Reply:
actually for the progress spinner, it is not a css style, it is a string property given to the `p-progressSpinner` component.

Reply:
fixed